pkg/skaffold/runner/dev.go

```diff
@@ -99,10 +104,23 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer) error {
 				eventV2.TaskFailed(constants.DevLoop, err)
 				endTrace(instrumentation.TraceEndError(err))
 
-				return nil
+				return err
 			}
 
 			fileSyncSucceeded(fileCount, s.Image)
+
+			return nil
+		}
+		for _, s := range r.changeSet.NeedsResync() {
+			err := backoff.Retry(
+				func() error {
+					return syncHandler(s)
+				}, backoff.WithContext(opts, childCtx),
+			)
+
+			if err != nil {
```

Comment:
why returning nil here?

Reply:
it's old logic, I left it as is. 
![image](https://github.com/GoogleContainerTools/skaffold/assets/3595194/6c9a955b-7ea2-458b-9797-88234871f43f)